### PR TITLE
Added Linux export symbol

### PR DESCRIPTION
**Pull request by locke-lunarg**
Added `.map` files to control which symbols should export or hide in Linux. The feature is the same with `.def` files in Visual Studio.

So, maybe we don't need `#define VK_LAYER_EXPORT __attribute__((visibility("default")))` any longer.

### CODE REVIEW

File: tests/layers/libVkLayer_device_profile_api.map [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  global:
```
Comment (jzulauf-lunarg) [inline tests/layers/libVkLayer_device_profile_api.map]: Why doesn't this one have vkGetDeviceProcAddr?

Comment (locke-lunarg) [reply tests/layers/libVkLayer_device_profile_api.map]: I just copy it from `VkLayer_device_profile_api.def`. It doesn't have `vkGetDeviceProcAddr` in `VkLayer_device_profile_api.def`.

Comment (locke-lunarg) [reply tests/layers/libVkLayer_device_profile_api.map]: https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/tests/layers/VkLayer_device_profile_api.def

**Comment by locke-lunarg** [comment]
> I don't think this is working. I still see a bunch of external symbols in the shared objects.
I use the commend to check the symbol. The commend is correct?
`nm -g -D -C --defined-only  libVkLayer_core_validation.so`

**Comment by jeremy-lunarg** [comment]
@locke-lunarg You're right, my mistake.

**Pull request merged**